tests/stub/iteration/scripts/v4x4/tx_pull_1_nested.script

```diff
@@ -9,18 +9,31 @@ S: SUCCESS {"fields": ["x"], "qid": 1}
 C: PULL {"n": 1, "[qid]": 1}
 S: RECORD ["1_1"]
    SUCCESS {"has_more": true}
-C: RUN "CYPHER" {} {}
-S: SUCCESS {"fields": ["x"], "qid": 2}
-C: PULL {"n": 1, "[qid]": 2}
-S: RECORD ["2_1"]
-   SUCCESS {"has_more": true}
-C: PULL {"n": 1, "[qid]": 2}
-S: RECORD ["2_2"]
-   SUCCESS {}
-C: PULL {"n": 1, "qid": 1}
-S: RECORD ["1_2"]
-   SUCCESS {}
-C: RUN "CYPHER" {} {}
+{{
+    C: PULL {"n": 1, "qid": 1}
+    S: RECORD ["1_2"]
+       SUCCESS {}
+    C: RUN "CYPHER NESTED" {} {}
+    S: SUCCESS {"fields": ["x"], "qid": 2}
+    C: PULL {"n": 1, "[qid]": 2}
+    S: RECORD ["2_1"]
+       SUCCESS {"has_more": true}
+    C: PULL {"n": 1, "[qid]": 2}
+    S: RECORD ["2_2"]
```

Comment:
I assume this whole alternative branch is to cater for drivers that don't pipeline a pull. Which makes perfect sense, but I have the feeling the test might need some restructuring because there is no nesting going on here. The two result streams are exhausted sequentially. First all records for qid 1, then all for qid 2.

Reply:
This also seems to apply for the other nested scripts.

Reply:
This is to accommodate drivers that implement a the low watermark logic that results in an immediate pull on first record consumption in this case. While this branch of execution in stub script is sequential, the test verifies that the driver can be used with nested queries.

Reply:
Sort of. I get the point, it's verifying that you can have two result objects living at the same time. But the test is not really forcing them to share the connection in an interleaved way, which I find worth having a test for. If you still don't think so, I'd consider this good enough.

Reply:
Updated.